pkg/cloud/aws/actuators/machine/actuator_test.go

```diff
@@ -195,38 +219,91 @@ func TestActuator(t *testing.T) {
 		name                    string
 		machine                 *clusterv1.Machine
 		error                   string
-		operation               func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine)
+		operation               func(client client.Client, actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine)
 		describeInstancesOutput *ec2.DescribeInstancesOutput
 		runInstancesErr         error
 		describeInstancesErr    error
 		terminateInstancesErr   error
 		lbErr                   error
 	}{
 		{
-			name:    "Update with success",
+			name:    "Create machine with success",
 			machine: machine,
-			operation: func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+			operation: func(objectClient client.Client, actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+				createErr := actuator.Create(context.TODO(), cluster, machine)
+				machineStatus, err := getMachineStatus(objectClient, machine)
+				if err != nil {
+					t.Fatalf("Unable to get machine status: %v", err)
+				}
+
+				assert.NoError(t, createErr)
```

Comment:
Should this happen closer to the call to `actuator.Create()`?

Reply:
At the end there is no difference, the `machineStatus` needs to be retrievable from the machine object no matter of the `Create` operation failed or not.

Reply:
If that's the case why not move it closer to the call. The pattern is call func, check for error in (really) most places. This stuck out, hence my comment/question.

Reply:
Done.